### PR TITLE
Add initial skeleton for catalog master

### PR DESCRIPTION
**Pull request by yuzhu**
Some skeleton code for the catalog master

### CODE REVIEW

**Comment by yuzhu** [comment]
@gpang @HelloHorizon Next I will add a skeleton for the client so that @HelloHorizon can make calls to the catalog master.

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by HelloHorizon** [comment]
@yuzhu should we add column statistics interface in master?
Like `public List<ColumnStatistics> getAllColumnStatistics(String databaseName, String tableName)`

**Comment by yuzhu** [comment]
> @yuzhu should we add column statistics interface in master?
> Like `public List<ColumnStatistics> getAllColumnStatistics(String databaseName, String tableName)`

will do

File: pom.xml [hunk 1]
```diff
@@ -233,6 +237,11 @@
         <artifactId>jnr-fuse</artifactId>
         <version>0.5.1</version>
       </dependency>
+      <dependency>
```
Comment (gpang) [inline pom.xml]: lets try to move the catalog into its own top-level module, and move all the dependencies to the lowest level possible.

File: core/base/src/main/java/alluxio/Constants.java [hunk 2]
```diff
@@ -94,6 +94,8 @@
   public static final long JOURNAL_MASTER_CLIENT_SERVICE_VERSION = 1;
   public static final long UNKNOWN_SERVICE_VERSION = -1;
 
+  public static final String CATALOG_MASTER_NAME = "CatalogMaster";
```
Comment (gpang) [inline core/base/src/main/java/alluxio/Constants.java]: can we move this to the other module?

File: core/server/master/pom.xml [hunk 3]
```diff
@@ -34,6 +34,10 @@
       <groupId>com.esotericsoftware</groupId>
       <artifactId>kryo</artifactId>
     </dependency>
+    <dependency>
```
Comment (gpang) [inline core/server/master/pom.xml]: can we remove this now?

File: pom.xml (hunk 1 above)
Comment (gpang) [reply pom.xml]: can we avoid modifying this top level pom?

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by gpang** [comment]
alluxio-bot, merge this please

**Pull request merged**